addons/calendar/models/calendar_event.py

```diff
@@ -401,16 +401,28 @@ def _inverse_dates(self):
                 # because fullcalendar just drops times for full day events.
                 # i.e. Christmas is on 25/12 for everyone
                 # even if people don't celebrate it simultaneously
-                enddate = fields.Datetime.from_string(meeting.stop_date)
-                enddate = enddate.replace(hour=18)
+                if meeting.start_date and meeting.stop_date:
+                    enddate = fields.Datetime.from_string(meeting.stop_date)
+                    enddate = enddate.replace(hour=18)
 
-                startdate = fields.Datetime.from_string(meeting.start_date)
-                startdate = startdate.replace(hour=8)  # Set 8 AM
+                    startdate = fields.Datetime.from_string(meeting.start_date)
+                    startdate = startdate.replace(hour=8)  # Set 8 AM
 
-                meeting.write({
-                    'start': startdate.replace(tzinfo=None),
-                    'stop': enddate.replace(tzinfo=None)
-                })
+                    meeting.write({
+                        'start': startdate.replace(tzinfo=None),
+                        'stop': enddate.replace(tzinfo=None)
+                    })
+                else:
+                    enddate = fields.Datetime.from_string(meeting.stop)
+                    enddate = enddate.replace(hour=18)
+
+                    startdate = fields.Datetime.from_string(meeting.start)
+                    startdate = startdate.replace(hour=8)  # Set 8 AM
+
+                    meeting.write({
+                        'start_date': startdate.replace(tzinfo=None),
+                        'stop_date': enddate.replace(tzinfo=None)
```

Comment:
Thanks for your work!!
We should add test case for this scenario like:
1.) create all day event
2.) now duplicate it and it should work without issue

 
Some code seems to be duplicated which can be avoided here also add a comments too. Also intend the  below suggested code.
```suggestion
        for meeting in self:
            if meeting.allday:

                # Convention break:
                # stop and start are NOT in UTC in allday event
                # in this case, they actually represent a date
                # because fullcalendar just drops times for full day events.
                # i.e. Christmas is on 25/12 for everyone
                # even if people don't celebrate it simultaneously
                startdate = fields.Datetime.from_string(meeting.start_date or meeting.start).replace(hour=8)
                enddate = fields.Datetime.from_string(meeting.stop_date or meeting.stop).replace(hour=18)

                if meeting.start_date and meeting.stop_date:
                    meeting.write({
                        'start': startdate.replace(tzinfo=None),
                        'stop': enddate.replace(tzinfo=None)
                    })
                else:
                    meeting.write({
                        'start_date': startdate.replace(tzinfo=None),
                        'stop_date': enddate.replace(tzinfo=None)
                    })
```